src/components/views/login/RegistrationForm.js

```diff
@@ -109,7 +110,8 @@ module.exports = React.createClass({
                     {emailSection}
                     <br />
                     <input className="mx_Login_field" type="text" ref="username"
-                        placeholder="User name" defaultValue={this.state.username} />
+                        placeholder="User name" defaultValue={this.state.username}
+                        disabled={this.props.disableUsernameChanges} />
```

Comment:
Yeah not sure on the best UX for that. I can hide instead of disable the username form, @ara4n ?